libcaf_net/caf/net/transport_worker.hpp

```diff
@@ -65,20 +81,16 @@ class transport_worker {
     application_.write_message(decorator, std::move(msg));
   }
 
-  template <class Parent>
-  void write_packet(Parent& parent, span<const byte> header,
```

Comment:
Why did you delete this function? Does the application need to get the `id_type` and pass it on to the transport itself?

Reply:
The only job of the decorator is to add this ID. That's why I changed `write_packet` in the decorator to avoid having all worker classes implement virtually identical `write_packet` functions. Here's the new code in the decorator:

```c++
  template <class... Ts>
  void write_packet(span<const byte> header, span<const byte> payload,
                    Ts&&... xs) {
    parent_.write_packet(header, payload, std::forward<Ts>(xs)...,
                         object_.id());
  }
```

Reply:
This does indeed seem cleaner. 👍

Reply:
Constant vigilance! 🧙‍♂️